client/extensions/woocommerce/state/action-list/reducer.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	WOOCOMMERCE_ACTION_LIST_CREATE,
+	WOOCOMMERCE_ACTION_LIST_CLEAR,
+	WOOCOMMERCE_ACTION_LIST_STEP_START,
+	WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS,
+	WOOCOMMERCE_ACTION_LIST_STEP_FAILURE,
+} from 'woocommerce/state/action-types';
+
+export default createReducer( null, {
+	[ WOOCOMMERCE_ACTION_LIST_CREATE ]: handleActionListCreate,
+	[ WOOCOMMERCE_ACTION_LIST_CLEAR ]: handleActionListClear,
+	[ WOOCOMMERCE_ACTION_LIST_STEP_START ]: handleActionListStepStart,
+	[ WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS ]: handleActionListStepSuccess,
+	[ WOOCOMMERCE_ACTION_LIST_STEP_FAILURE ]: handleActionListStepFailure,
+} );
+
+function handleActionListCreate( actionList, action ) {
+	// The action list given in the action is our new list.
+	// TODO: validate action list?
+	return action.actionList;
+}
+
+function handleActionListClear() {
+	// Clear out the existing action list.
+	return null;
+}
+
+function handleActionListStepStart( actionList, action ) {
+	const { stepIndex, time } = action;
+	const startTime = time || Date.now();
```

Comment:
So, this is actually solved in a later PR when I switch to the reducer using annotation actions. Then the Date.now() is set by a data-layer handler: https://github.com/Automattic/wp-calypso/pull/15090/files#diff-ce926988aaba4ce8dd45684e5aa6956bL32